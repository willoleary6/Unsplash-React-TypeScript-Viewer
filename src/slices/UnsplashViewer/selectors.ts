import { RootState } from "../../app/store";
import { UnsplashViewerState } from "./slice";

export const selectUnsplashViewer = (state: RootState): UnsplashViewerState => state.unsplash;
