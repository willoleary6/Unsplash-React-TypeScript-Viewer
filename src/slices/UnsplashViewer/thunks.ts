import { createAsyncThunk } from "@reduxjs/toolkit";

export const searchUnsplashData = createAsyncThunk(
    "unsplash/searchUnsplash",
    async (searchInput: string) => {
        const response = await fetch(
            "https://api.unsplash.com/search/photos?query=" +
                searchInput +
                "&client_id=M5VBH_2QX33zPVyoWGvnVmb8zQbcESVOyCqQO185FEk",
            {
                method: "GET",
            }
        )
            .then((response) => {
                if (!response.ok) {
                    throw Error(response.statusText);
                }
                return response.json();
            })
            .catch(() => {
                return null;
            });
        return response;
    }
);
