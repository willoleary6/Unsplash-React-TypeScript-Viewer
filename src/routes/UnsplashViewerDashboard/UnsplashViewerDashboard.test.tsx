import React from "react";
import { UnsplashViewerDashboard } from "./UnsplashViewerDashboard";
import Enzyme, { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import configureStore from "redux-mock-store";
import thunk, { ThunkDispatch } from "redux-thunk";
import { AnyAction } from "redux";
import { UnsplashViewerState, initialState } from "../../slices/UnsplashViewer/slice";
import { Provider } from "react-redux";

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
interface mockStoreInterface {
    unsplash: UnsplashViewerState;
}
const mockStore = configureStore<
    mockStoreInterface,
    ThunkDispatch<mockStoreInterface, unknown, AnyAction>
>(middlewares);

describe("GalleryModal render testing", () => {
    let store = mockStore();

    it("renders and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        // using mount here instead of shallow because we want to render the child components too
        const mockState: mockStoreInterface = {
            unsplash: { ...initialState, searchStatus: "Idle" },
        };
        store = mockStore(mockState);
        const wrapper = mount(
            <React.StrictMode>
                <Provider store={store}>
                    <UnsplashViewerDashboard />
                </Provider>
            </React.StrictMode>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders and takes snapshot to track jsx element changes, should render the loading animation", () => {
        const mockState: mockStoreInterface = {
            unsplash: { ...initialState, searchStatus: "In Progress" },
        };
        store = mockStore(mockState);
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = mount(
            <React.StrictMode>
                <Provider store={store}>
                    <UnsplashViewerDashboard />
                </Provider>
            </React.StrictMode>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders and takes snapshot to track jsx element changes, should render the gallery component", () => {
        const mockState: mockStoreInterface = {
            unsplash: { ...initialState, searchStatus: "Success" },
        };
        store = mockStore(mockState);
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = mount(
            <React.StrictMode>
                <Provider store={store}>
                    <UnsplashViewerDashboard />
                </Provider>
            </React.StrictMode>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders and takes snapshot to track jsx element changes, should render the Error message ", () => {
        const mockState: mockStoreInterface = {
            unsplash: { ...initialState, searchStatus: "Error" },
        };
        store = mockStore(mockState);
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = mount(
            <React.StrictMode>
                <Provider store={store}>
                    <UnsplashViewerDashboard />
                </Provider>
            </React.StrictMode>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders and takes snapshot to track jsx element changes, should render the All results displayed image", () => {
        const mockState: mockStoreInterface = {
            unsplash: { ...initialState, searchStatus: "All results retrieved" },
        };
        store = mockStore(mockState);
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = mount(
            <React.StrictMode>
                <Provider store={store}>
                    <UnsplashViewerDashboard />
                </Provider>
            </React.StrictMode>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders and takes snapshot to track jsx element changes, should render the No Results message", () => {
        const mockState: mockStoreInterface = {
            unsplash: { ...initialState, searchStatus: "No Results" },
        };
        store = mockStore(mockState);
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = mount(
            <React.StrictMode>
                <Provider store={store}>
                    <UnsplashViewerDashboard />
                </Provider>
            </React.StrictMode>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
