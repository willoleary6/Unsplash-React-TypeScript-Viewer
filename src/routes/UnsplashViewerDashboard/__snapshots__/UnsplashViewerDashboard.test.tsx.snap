// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryModal render testing renders and takes snapshot to track jsx element changes, should render the All results displayed image 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UnsplashViewerDashboard>
    <div
      className="row"
    >
      <div
        className="col-lg-4 mt-5"
      />
      <div
        className="col-lg-4 mt-5"
      >
        <SearchBar
          executeSearchFunction={[Function]}
        >
          <div
            className="flex justify-center"
          >
            <div
              className="mb-3 xl:w-96"
            >
              <div
                className="input-group relative flex flex-wrap items-stretch w-full mb-4"
              >
                <input
                  aria-describedby="button-addon2"
                  aria-label="Search"
                  className="form-control relative flex-auto min-w-0 block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search"
                  type="search"
                />
                <button
                  className="btn px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700  focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out flex items-center"
                  disabled={true}
                  id="button-addon2"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    className="w-4"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </SearchBar>
      </div>
      <div
        className="col-lg-4 mt-5"
      />
    </div>
    <Gallery>
      <section
        className="overflow-hidden text-gray-700  flex flex-col"
      >
        <div
          className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32"
        >
          <InfiniteScroll
            className="flex flex-wrap -m-1 md:-m-2 overflow-hidden "
            dataLength={0}
            hasMore={false}
            loader={<LoadingSpinner />}
            next={[Function]}
          >
            <div
              className="infinite-scroll-component__outerdiv"
              style={Object {}}
            >
              <div
                className="infinite-scroll-component flex flex-wrap -m-1 md:-m-2 overflow-hidden "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": "auto",
                    "overflow": "auto",
                  }
                }
              />
            </div>
          </InfiniteScroll>
        </div>
        <GalleryModal
          setShowModal={[Function]}
          showModal={false}
        />
      </section>
    </Gallery>
    <SearchStatusBox
      isError={false}
      messageToDisplay="All results retrieved"
    >
      <div
        className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32 text-center bg-gray-300"
      >
        <p>
          All results retrieved
        </p>
      </div>
    </SearchStatusBox>
  </UnsplashViewerDashboard>
</Provider>
`;

exports[`GalleryModal render testing renders and takes snapshot to track jsx element changes, should render the Error message  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UnsplashViewerDashboard>
    <div
      className="row"
    >
      <div
        className="col-lg-4 mt-5"
      />
      <div
        className="col-lg-4 mt-5"
      >
        <SearchBar
          executeSearchFunction={[Function]}
        >
          <div
            className="flex justify-center"
          >
            <div
              className="mb-3 xl:w-96"
            >
              <div
                className="input-group relative flex flex-wrap items-stretch w-full mb-4"
              >
                <input
                  aria-describedby="button-addon2"
                  aria-label="Search"
                  className="form-control relative flex-auto min-w-0 block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search"
                  type="search"
                />
                <button
                  className="btn px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700  focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out flex items-center"
                  disabled={true}
                  id="button-addon2"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    className="w-4"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </SearchBar>
      </div>
      <div
        className="col-lg-4 mt-5"
      />
    </div>
    <Gallery>
      <section
        className="overflow-hidden text-gray-700  flex flex-col"
      >
        <div
          className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32"
        >
          <InfiniteScroll
            className="flex flex-wrap -m-1 md:-m-2 overflow-hidden "
            dataLength={0}
            hasMore={false}
            loader={<LoadingSpinner />}
            next={[Function]}
          >
            <div
              className="infinite-scroll-component__outerdiv"
              style={Object {}}
            >
              <div
                className="infinite-scroll-component flex flex-wrap -m-1 md:-m-2 overflow-hidden "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": "auto",
                    "overflow": "auto",
                  }
                }
              />
            </div>
          </InfiniteScroll>
        </div>
        <GalleryModal
          setShowModal={[Function]}
          showModal={false}
        />
      </section>
    </Gallery>
    <SearchStatusBox
      isError={true}
      messageToDisplay="Something went wrong...."
    >
      <div
        className="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative text-center"
        role="alert"
      >
        <span
          className="block sm:inline"
        >
          Something went wrong....
        </span>
      </div>
    </SearchStatusBox>
  </UnsplashViewerDashboard>
</Provider>
`;

exports[`GalleryModal render testing renders and takes snapshot to track jsx element changes, should render the No Results message 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UnsplashViewerDashboard>
    <div
      className="row"
    >
      <div
        className="col-lg-4 mt-5"
      />
      <div
        className="col-lg-4 mt-5"
      >
        <SearchBar
          executeSearchFunction={[Function]}
        >
          <div
            className="flex justify-center"
          >
            <div
              className="mb-3 xl:w-96"
            >
              <div
                className="input-group relative flex flex-wrap items-stretch w-full mb-4"
              >
                <input
                  aria-describedby="button-addon2"
                  aria-label="Search"
                  className="form-control relative flex-auto min-w-0 block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search"
                  type="search"
                />
                <button
                  className="btn px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700  focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out flex items-center"
                  disabled={true}
                  id="button-addon2"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    className="w-4"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </SearchBar>
      </div>
      <div
        className="col-lg-4 mt-5"
      />
    </div>
    <Gallery>
      <section
        className="overflow-hidden text-gray-700  flex flex-col"
      >
        <div
          className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32"
        >
          <InfiniteScroll
            className="flex flex-wrap -m-1 md:-m-2 overflow-hidden "
            dataLength={0}
            hasMore={false}
            loader={<LoadingSpinner />}
            next={[Function]}
          >
            <div
              className="infinite-scroll-component__outerdiv"
              style={Object {}}
            >
              <div
                className="infinite-scroll-component flex flex-wrap -m-1 md:-m-2 overflow-hidden "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": "auto",
                    "overflow": "auto",
                  }
                }
              />
            </div>
          </InfiniteScroll>
        </div>
        <GalleryModal
          setShowModal={[Function]}
          showModal={false}
        />
      </section>
    </Gallery>
    <SearchStatusBox
      isError={false}
      messageToDisplay="Unable to find any images for that query"
    >
      <div
        className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32 text-center bg-gray-300"
      >
        <p>
          Unable to find any images for that query
        </p>
      </div>
    </SearchStatusBox>
  </UnsplashViewerDashboard>
</Provider>
`;

exports[`GalleryModal render testing renders and takes snapshot to track jsx element changes, should render the gallery component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UnsplashViewerDashboard>
    <div
      className="row"
    >
      <div
        className="col-lg-4 mt-5"
      />
      <div
        className="col-lg-4 mt-5"
      >
        <SearchBar
          executeSearchFunction={[Function]}
        >
          <div
            className="flex justify-center"
          >
            <div
              className="mb-3 xl:w-96"
            >
              <div
                className="input-group relative flex flex-wrap items-stretch w-full mb-4"
              >
                <input
                  aria-describedby="button-addon2"
                  aria-label="Search"
                  className="form-control relative flex-auto min-w-0 block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search"
                  type="search"
                />
                <button
                  className="btn px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700  focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out flex items-center"
                  disabled={true}
                  id="button-addon2"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    className="w-4"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </SearchBar>
      </div>
      <div
        className="col-lg-4 mt-5"
      />
    </div>
    <Gallery>
      <section
        className="overflow-hidden text-gray-700  flex flex-col"
      >
        <div
          className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32"
        >
          <InfiniteScroll
            className="flex flex-wrap -m-1 md:-m-2 overflow-hidden "
            dataLength={0}
            hasMore={true}
            loader={<LoadingSpinner />}
            next={[Function]}
          >
            <div
              className="infinite-scroll-component__outerdiv"
              style={Object {}}
            >
              <div
                className="infinite-scroll-component flex flex-wrap -m-1 md:-m-2 overflow-hidden "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": "auto",
                    "overflow": "auto",
                  }
                }
              >
                <LoadingSpinner>
                  <div
                    className=" grid place-items-center"
                  >
                    <svg
                      className="inline w-10 h-10 mr-2 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600 "
                      fill="none"
                      role="status"
                      viewBox="0 0 100 101"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
                        fill="currentColor"
                      />
                      <path
                        d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
                        fill="currentFill"
                      />
                    </svg>
                  </div>
                </LoadingSpinner>
              </div>
            </div>
          </InfiniteScroll>
        </div>
        <GalleryModal
          setShowModal={[Function]}
          showModal={false}
        />
      </section>
    </Gallery>
  </UnsplashViewerDashboard>
</Provider>
`;

exports[`GalleryModal render testing renders and takes snapshot to track jsx element changes, should render the loading animation 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UnsplashViewerDashboard>
    <div
      className="row"
    >
      <div
        className="col-lg-4 mt-5"
      />
      <div
        className="col-lg-4 mt-5"
      >
        <SearchBar
          executeSearchFunction={[Function]}
        >
          <div
            className="flex justify-center"
          >
            <div
              className="mb-3 xl:w-96"
            >
              <div
                className="input-group relative flex flex-wrap items-stretch w-full mb-4"
              >
                <input
                  aria-describedby="button-addon2"
                  aria-label="Search"
                  className="form-control relative flex-auto min-w-0 block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search"
                  type="search"
                />
                <button
                  className="btn px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700  focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out flex items-center"
                  disabled={true}
                  id="button-addon2"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    className="w-4"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </SearchBar>
      </div>
      <div
        className="col-lg-4 mt-5"
      />
    </div>
    <Gallery>
      <section
        className="overflow-hidden text-gray-700  flex flex-col"
      >
        <div
          className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32"
        >
          <InfiniteScroll
            className="flex flex-wrap -m-1 md:-m-2 overflow-hidden "
            dataLength={0}
            hasMore={false}
            loader={<LoadingSpinner />}
            next={[Function]}
          >
            <div
              className="infinite-scroll-component__outerdiv"
              style={Object {}}
            >
              <div
                className="infinite-scroll-component flex flex-wrap -m-1 md:-m-2 overflow-hidden "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": "auto",
                    "overflow": "auto",
                  }
                }
              />
            </div>
          </InfiniteScroll>
        </div>
        <GalleryModal
          setShowModal={[Function]}
          showModal={false}
        />
      </section>
    </Gallery>
    <LoadingSpinner>
      <div
        className=" grid place-items-center"
      >
        <svg
          className="inline w-10 h-10 mr-2 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600 "
          fill="none"
          role="status"
          viewBox="0 0 100 101"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
            fill="currentColor"
          />
          <path
            d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
            fill="currentFill"
          />
        </svg>
      </div>
    </LoadingSpinner>
  </UnsplashViewerDashboard>
</Provider>
`;

exports[`GalleryModal render testing renders and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UnsplashViewerDashboard>
    <div
      className="row"
    >
      <div
        className="col-lg-4 mt-5"
      />
      <div
        className="col-lg-4 mt-5"
      >
        <SearchBar
          executeSearchFunction={[Function]}
        >
          <div
            className="flex justify-center"
          >
            <div
              className="mb-3 xl:w-96"
            >
              <div
                className="input-group relative flex flex-wrap items-stretch w-full mb-4"
              >
                <input
                  aria-describedby="button-addon2"
                  aria-label="Search"
                  className="form-control relative flex-auto min-w-0 block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search"
                  type="search"
                />
                <button
                  className="btn px-6 py-2.5 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700  focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out flex items-center"
                  disabled={true}
                  id="button-addon2"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    className="w-4"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </SearchBar>
      </div>
      <div
        className="col-lg-4 mt-5"
      />
    </div>
    <Gallery>
      <section
        className="overflow-hidden text-gray-700  flex flex-col"
      >
        <div
          className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32"
        >
          <InfiniteScroll
            className="flex flex-wrap -m-1 md:-m-2 overflow-hidden "
            dataLength={0}
            hasMore={false}
            loader={<LoadingSpinner />}
            next={[Function]}
          >
            <div
              className="infinite-scroll-component__outerdiv"
              style={Object {}}
            >
              <div
                className="infinite-scroll-component flex flex-wrap -m-1 md:-m-2 overflow-hidden "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": "auto",
                    "overflow": "auto",
                  }
                }
              />
            </div>
          </InfiniteScroll>
        </div>
        <GalleryModal
          setShowModal={[Function]}
          showModal={false}
        />
      </section>
    </Gallery>
    <SearchStatusBox
      isError={false}
      messageToDisplay="Search and view a gallery of images provided by Unsplash"
    >
      <div
        className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32 text-center bg-gray-300"
      >
        <p>
          Search and view a gallery of images provided by Unsplash
        </p>
      </div>
    </SearchStatusBox>
  </UnsplashViewerDashboard>
</Provider>
`;
